src/providers/pkcs11/key_management.rs

```diff
@@ -305,11 +361,48 @@ impl Provider {
             modulus,
             public_exponent,
         };
-        let data = picky_asn1_der::to_vec(&key).map_err(|err| {
+        Ok(picky_asn1_der::to_vec(&key).map_err(|err| {
             format_error!("Could not serialise key elements", err);
             ResponseStatus::PsaErrorCommunicationFailure
-        })?;
-        Ok(psa_export_public_key::Result { data: data.into() })
+        })?)
+    }
+
+    fn export_public_ec_internal(
+        &self,
+        key: ObjectHandle,
+        session: &Session,
+        key_attributes: Attributes,
+    ) -> Result<Vec<u8>> {
+        let mut attributes = session
+            .get_attributes(key, &[AttributeType::EcPoint])
+            .map_err(to_response_status)?;
+
+        if attributes.len() != 1 {
+            error!("Expected to find EC point attribute in public key.");
+            return Err(ResponseStatus::PsaErrorCommunicationFailure);
+        }
```

Comment:
Maybe this is already covered in the `else` arm of line 385?

Reply:
If you call `remove` on a vector and the index is out of bounds, you get a panic ☠️